### PR TITLE
Added a makefile for building and testing.

### PR DESCRIPTION
**Pull request by amirmghaemi**
This is a small change, but we can use the makefile for authentication and/or starting the log generating bots in the future as well. Also added the `make build` and `make test` commands to the README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -301,11 +301,22 @@ you will need to rebuild the UserALE.js script (and run tests).
...
 ```
 
 To run UserALE.js unit tests:
 ```
+make test
+```
+Alternatively,
+```
+npm install
+npm run build
```
Comment (Jyyjy) [inline README.md:319]: No need to build before testing, `npm run test` will clean and rebuild

File: makefile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+.PHONY: build test
+
+build:
+	npm install
+	npm run build
+test:
+	npm install
+	npm run build
```
Comment (Jyyjy) [inline makefile:8]: Same thing here.

**Review by Jyyjy — APPROVED** [review]
Looks good, as long as the make file is essentially aliases for npm commands. Almost everything that can done in a make file, can be done with npm commands. And some people (me) are still going to use npm commands.

So in order for everyone to be on the same page, new functionality should added to package.json scripts and not the makefile.

**Comment by EandrewJones** [comment]
I know this is ready to merge, but I'm a strong NO on adding a makefile unless it's solving a problem that cannot be handled by the standard javascript tooling (npm, yarn, npx, ...). In our case, it is not.

**Pull request closed without merging**